version: '3.8'

services:
    db:
        image: postgres:14.4
        volumes:
            - postgres_data:/var/lib/postgresql/data
        ports:
            - "5432:5432"
        environment:
            POSTGRES_DB: postgres
            POSTGRES_USER: postgres
            POSTGRES_PASSWORD: postgres
        healthcheck:
            test: ["CMD-SHELL", "pg_isready -U postgres"]
            interval: 10s
            timeout: 5s
            retries: 5

    web:
        build:
            context: .
            dockerfile: Dockerfile
        volumes:
            - .:/app
        ports:
            - "8000:8000"
        depends_on:
            - db
        restart: on-failure
        command: ["python", "manage.py", "runserver", "0.0.0.0:8000"]

volumes:
    postgres_data:
